libsolidity/ast/AST.cpp

```diff
@@ -338,6 +338,7 @@ shared_ptr<FunctionType> FunctionDefinition::functionType(bool _internal) const
 
 TypePointer FunctionDefinition::type() const
 {
+	solAssert(visibility() != Declaration::Visibility::External, "");
```

Comment:
Why is this needed?

Reply:
That's a good question. I don't remember, but it does not seem to fail...

Reply:
Ah, the `FunctionType` constructor has a second argument that tells whether we want to have the type of the internal aspect or the type of the external aspect. It defaults to "internal".